etc/migration/10.3-to-11.0/mapped-emails.js

```diff
@@ -0,0 +1,122 @@
+ * A CSV file with all errors will be created in the current working directory. If any of the above
+ * checks fail, you will have to manually address these. How you resolve these issues is up to you.
+ *
+ * Once you've addressed the raised issues, re-run the script to perform the actual migration
+ */
+
+var _ = require('underscore');
+var bunyan = require('bunyan');
+var csv = require('csv');
+var fs = require('fs');
+
+var Cassandra = require('oae-util/lib/cassandra');
+var log = require('oae-logger').logger('revisions-migrator');
+var OAE = require('oae-util/lib/oae');
+var PrincipalsDAO = require('oae-principals/lib/internal/dao');
+var Validator = require('oae-util/lib/validator').Validator;
+
+// The application configuration
+var config = require('../../../config').config;
+
+// Ensure that this application server does NOT start processing any preview images
+config.previews.enabled = false;
+
+// Ensure that we're logging to standard out/err
+config.log = {
+    'streams': [
+        {
+            'level': 'info',
+            'stream': process.stdout
+        }
+    ]
+};
+
+// Keep track of when we started the migration process so we can output how
+// long the migration took
+var start = Date.now();
+
+// Keep track of the total number of users and the number of users that were mapped
+var totalUsers = 0;
+var mappedUsers = 0;
+
+// Spin up the application container. This will allow us to re-use existing APIs
+OAE.init(config, function(err) {
+    if (err) {
+        log().error({'err': err}, 'Unable to spin up the application server');
+        return _exit(err.code);
+    }
+
+    PrincipalsDAO.iterateAll(['principalId', 'displayName', 'email'], 30, _mapPrincipals, function() {
+        if (err) {
+            log().error({'err': err}, 'Failed to migrate all users');
+            process.exit(1);
+        }
+
+        log().info('Migration completed, it took %d milliseconds', (Date.now() - start));
+        log().info('Total users: %d, mapped users: %d', totalUsers, mappedUsers);
+        process.exit(0);
+    });
+});
+
+/**
+ * Go through a set of rows and map the users who have a valid email address
+ *
+ * @param  {Object[]}   principals      An array of principal objects to map
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error object, if any
+ * @api private
+ */
+var _mapPrincipals = function(principals, callback) {
+    var queries = [];
+
+    _.each(principals, function(principal) {
+        var principalId = principal.principalId;
+        var displayName = principal.displayName;
+        var email = principal.email;
+
+        // We only care about users in this migration process
+        if (!PrincipalsDAO.isUser(principalId)) {
+            return;
+        }
+        totalUsers++;
+
+        // Check if the user has an email address
+        if (!email) {
+            return log().warn({'principalId': principalId}, 'This user has no email address');
```

Comment:
If we're rejecting the mapping here, we should probably keep track of why it was rejected (e.g., a "noEmail" count)